python/ray/serve/handle.py

```diff
@@ -60,17 +108,15 @@ def remote(self, request_data: Optional[Union[Dict, Any]] = None,
             ``**kwargs``: All keyword arguments will be available in
                 ``request.args``.
         """
-        request_metadata = RequestMetadata(
-            get_random_letters(10),  # Used for debugging.
-            self.endpoint_name,
-            TaskContext.Python,
-            call_method=self.method_name or "__call__",
-            shard_key=self.shard_key,
-            http_method=self.http_method or "GET",
-            http_headers=self.http_headers or dict(),
-        )
-        return self.router_handle.enqueue_request.remote(
-            request_metadata, request_data, **kwargs)
+        assert self.sync, "handle.remote() should be called from sync handle."
+        coro = self._remote(request_data, kwargs)
+        future = asyncio.run_coroutine_threadsafe(coro, self.async_loop)
+        return future.result()
```

Comment:
Hmm, this doesn't seem like the interface we want -- this will throw an exception if the future's result isn't yet availabe

Reply:
Actually, this future is a concurrent.futures.Future object, which will block until the result is ready when you call `future.result()`.

Reply:
Wow, that's quite confusing...